backend/tests/integration/test_service_integration.py

```
@@ -0,0 +1,512 @@
+            # Create dataset
+            CommandResult(success=True, returncode=0, stdout="", stderr=""),
+            # Get dataset (for verification)
+            CommandResult(success=True, returncode=0, stdout="pool1/dataset1\t1G\t500M\t500M\tlz4\tsha256", stderr=""),
+            # List datasets
+            CommandResult(success=True, returncode=0, stdout="pool1/dataset1\t1G\t500M\t500M\tlz4\tsha256", stderr=""),
+            # Delete dataset
+            CommandResult(success=True, returncode=0, stdout="", stderr="")
+        ]
+        
+        # Test create dataset
+        create_result = legacy_adapter.create_dataset("pool1/dataset1", {"compression": "lz4"})
```

Ruff (0.11.9)

Undefined name `legacy_adapter`

(F821)